### ISSUE TITLE
the performance dramatically decreases when add Chinese bert

### ISSUE THREAD

**Issue by yzho0907** [WindChimeRan/pytorch_multi_head_selection_re#6]
i was trying to test perfomance of this model when train it by using baidu data but after fix few bugs:
lack of bert_tokenizer.tokenize(text) when using pad_bert in selection_loader.py
sequence_tags sometimes out of range when using find_entity in selection.py

the code is finally to runnable for using Chinese bert(pretrain_model has been changed to Chinese bert and vocab.txt as well), but after 20 epochs, i try to evaluation:
Triplets-> p: 0.4542, r: 0.2799, f: 0.3464 ||NER->p: 0.8442, r: 0.7668, f: 0.8036

 any clue i might get it incorrect? thx

**Comment by WindChimeRan** [comment]
Actually there are some strange things in CoNLL-BERT as well.
The phenomenon is:
The gradient would explode if I continue training all the layers in BERT. 
My remedy: only finetune the last layer of BERT.

My friend told me that he tried Chinese-BERT as well. The model works great and F1-triplet is around 0.80.

**Comment by yzho0907** [comment]
thx. 
Strange indeed, if the gradient explodes, it will only lead to the model could not coverage but the loss seems still decreasing(NER seems fine as well maybe something wrong in Triplets loss)

**Comment by IT-Xue** [comment]
i got this problem too,
i try to frozen all the layers of bert and add three lstm layers, this problem still exist.
Maybe the bert encoder have something wrong?

**Comment by yzho0907** [comment]
@Arthassssss sry, we haven't stick this repo anymore but i think the isuue is not from the gradient or layers in your models(cuz the loss was still decreasing).

**Comment by ghost** [comment]
> i got this problem too,
> i try to frozen all the layers of bert and add three lstm layers, this problem still exist.
> Maybe the bert encoder have something wrong?



> @Arthassssss sry, we haven't stick this repo anymore but i think the isuue is not from the gradient or layers in your models(cuz the loss was still decreasing).

Do you have a solution?

**Comment by yzho0907** [comment]
@illcat should try to change some hyperparams first: check the optimization methods and learning rate when bert involves (freeze all layers in bert first)

**Comment by moyan007** [comment]
Do you have solved the problem？I also got the problem when use bert as the encoder.

**Comment by moyan007** [comment]
> i was trying to test perfomance of this model when train it by using baidu data but after fix few bugs:
> lack of bert_tokenizer.tokenize(text) when using pad_bert in selection_loader.py
> sequence_tags sometimes out of range when using find_entity in selection.py
> 
> the code is finally to runnable for using Chinese bert(pretrain_model has been changed to Chinese bert and vocab.txt as well), but after 20 epochs, i try to evaluation:
> Triplets-> p: 0.4542, r: 0.2799, f: 0.3464 ||NER->p: 0.8442, r: 0.7668, f: 0.8036
> 
> any clue i might get it incorrect? thx

Do you have solved the problem? Could please give some advices when use bert as the encoder?

**Comment by yzho0907** [comment]
@moyan007 No, but i recommend u to check and try bert4keras, keras_bert, which suport Chinese model and there are many relation extraction examples(for spo datasets).i.e.https://github.com/cdjasonj/CCF_IE

**Comment by sfijfl** [comment]
i got another problem when train it by using baidu data，F1-triplet is around 0.0，but triple_loss,ner_loss,F1-ner is normal. i found that this is caused by the evaluation function，after fixing this bug，taking into account that the bottom layer bert has been pre-trained, the upper layer multi head is trained from scratch, so a larger training rate is used.
The model works great ,after 4 epochs,i try to evaluate: 
Triplets-> p: 0.7895, r: 0.8174, f: 0.8032 ||NER->p: 0.9082, r: 0.9369, f: 0.9223
bert learning_rate: 3e-5, upper_learning_rate: 1e-3

**Comment by yzho0907** [comment]
@sfijfl would you release your evaluation function here to help close this issue? thx a lot.
